sql/src/parser.rs

```diff
@@ -352,8 +352,39 @@ impl<'a> Parser<'a> {
                 );
             }
         }
+        let mut cols: Vec<ColumnDef> = vec![];
+        for mut column in columns {
+            let options = column.options;
+            let mut new_opts = vec![];
+            for option in options {
+                match &option.option {
+                    ColumnOption::DialectSpecific(ts) => {
+                        if ts.is_empty() || ts.len() != 1 {
+                            continue;
+                        }
+                        let token = &ts[0];
+                        match token {
+                            Token::Word(w) if w.value.eq(TS_KEY) => {
+                                constraints.push(TableConstraint::Unique {
+                                    name: Some(Ident {
+                                        value: TS_KEY.to_owned(),
+                                        quote_style: None,
+                                    }),
+                                    columns: vec![column.name.clone()],
+                                    is_primary: false,
+                                });
+                            }
+                            t => warn!("Unsupported Keyword in column option: {}", t),
+                        }
+                    }
+                    _ => new_opts.push(option),
+                }
+            }
+            column.options = new_opts;
+            cols.push(column);
```

Comment:
I think we can put this procedure as a method called `try_build_timestamp_key_constraint` for the simplicity of the `parse_columns`.

The method may looks like:
```rust
fn try_build_tskey_constraint(cols: &[ColumnDef]) -> Option<TableConstraint> {
...
}
```